tweets/views.py

```diff
@@ -36,3 +55,44 @@ class TweetDeleteView(LoginRequiredMixin, UserPassesTestMixin, DeleteView):
 
     def test_func(self):
         return self.request.user == self.get_object().user
+
+
+class LikeView(LoginRequiredMixin, View):
+    def post(self, request, *args, **kwargs):
+        tweet_id = self.kwargs["pk"]
+        tweet = get_object_or_404(Tweet, pk=tweet_id)
+        user = self.request.user
+        Like.objects.get_or_create(tweet=tweet, user=user)
+        is_liked = True
+        like_url = reverse("tweets:like", kwargs={"pk": tweet_id})
+        unlike_url = reverse("tweets:unlike", kwargs={"pk": tweet_id})
+        like_count = tweet.liked_tweet.count()
+        context = {
+            "like_count": like_count,
+            "tweet_id": tweet_id,
+            "is_liked": is_liked,
+            "like_url": like_url,
+            "unlike_url": unlike_url,
+        }
+        return JsonResponse(context)
+
+
+class UnlikeView(LoginRequiredMixin, View):
+    def post(self, request, *args, **kwargs):
+        tweet_id = self.kwargs["pk"]
+        tweet = get_object_or_404(Tweet, pk=tweet_id)
+        user = self.request.user
+        if like := Like.objects.filter(user=user, tweet=tweet):
+            like.delete()

```

Comment:
```suggestion
        Like.objects.filter(user=user, tweet=tweet).delete()
```
空のクエリセットに対してdeleteしてもエラーを出さないことを利用して、このように書くこともできます（修正の必要はないです）。